addons/pos_stripe/static/src/js/payment_stripe.js

```diff
@@ -0,0 +1,201 @@
+            }
+        } else if (this.terminal.getConnectionStatus() == 'not_connected') {
+            return await this.connectReader();
+        } else {
+            return true;
+        }
+    },
+
+    connectReader: async function () {
+        let line = this.pos.get_order().selected_paymentline;
+        let discoveredReaders = JSON.parse(this.pos.discoveredReaders);
+        for (const selectedReader of discoveredReaders) {
+            if (selectedReader.serial_number == this.payment_method.stripe_serial_number) {
+                let connectResult = await this.terminal.connectReader(selectedReader, {fail_if_in_use: true});
+                if (connectResult.error) {
+                    this._showError(connectResult.error.message, connectResult.error.code);
+                    line.set_payment_status('retry');
+                    return false;
+                } else {
+                    this.pos.connectedReader = this.payment_method.stripe_serial_number;
+                    return true;
+                }
+            }
+        }
+    },
+
+    collectPayment: async function (currency, amount) {
+        let line = this.pos.get_order().selected_paymentline;
+        let clientSecret = await this.fetchPaymentIntentClientSecret(currency, amount);
+        line.set_payment_status('waitingCard');
+        let collectPaymentMethod = await this.terminal.collectPaymentMethod(clientSecret);
+        if (collectPaymentMethod.error) {
+            this._showError(collectPaymentMethod.error.message, collectPaymentMethod.error.code);
+            line.set_payment_status('retry');
+            return false;
+        } else {
+            line.set_payment_status('waitingCapture');
+            let processPayment = await this.terminal.processPayment(collectPaymentMethod.paymentIntent);
+            if (processPayment.error) {
+                this._showError(processPayment.error.message, processPayment.error.code);
+                line.set_payment_status('retry');
+                return false;
+            } else if (processPayment.paymentIntent) {
+                line.set_payment_status('waitingCapture');
+                let capturePayment = await this.capturePayment(processPayment.paymentIntent.id);
+                line.set_payment_status('done');
+                line.transaction_id = (capturePayment.id);
+                return true;
+            }
+        }
+    },
+
+    capturePayment: function (paymentIntentId) {
+        return rpc.query({
+            model: 'pos.payment.method',
+            method: 'stripe_capture_payment',
+            args: [paymentIntentId],
+        }, {
+            silent: true,
+        }).catch(function (error) {
+            this._showError(_t('error'));
+        });
+    },
+
+    fetchPaymentIntentClientSecret: function (currency, amount) {
+        return rpc.query({
+            model: 'pos.payment.method',
+            method: 'stripe_payment_intent',
+            args: [currency, amount],
```

Comment:
so here if you are not in api.model anymore, and have passed the payment_method_id. You should have something like args: [[payment_method_id], currency, amount]